src/utilities.ts

```diff
@@ -43,6 +44,50 @@ export function getMinorFullName(acronym: string): string {
   return requirementJSON.minor[acronym].name;
 }
 
+export function getAllSubjects(): ReadonlySet<string> {
+  const set = new Set<string>();
+  fullCoursesArray.forEach(it => set.add(it.subject));
+  return set;
+}
+
+const SUBJECT_COLORS = [
+  {
+    text: 'Red',
+    hex: 'DA4A4A',
+  },
+  {
+    text: 'Orange',
+    hex: 'FFA53C',
+  },
+  {
+    text: 'Green',
+    hex: '58C913',
+  },
+  {
+    text: 'Blue',
+    hex: '139DC9',
+  },
+  {
+    text: 'Purple',
+    hex: 'C478FF',
+  },
+  {
+    text: 'Pink',
+    hex: 'F296D3',
+  },
+];
```

Comment:
Good catch for the dupe info @benjamin-shen. Why do you think this would be a breaking change?

Reply:
`colors.ts` stores hexes as `#FFFFFF`, with a hashtag. The hexes in this PR and the hexes on firestore are like `FFFFFF`, without a hashtag. So, when firestore data is converted to color styling, a hashtag is added (see `Course.vue` for an example). To write a color (i.e. due to a user color change) to the database, the hashtagged hex code is first substringed (see `CourseMenu.vue`).

So I initially thought it would be a good idea to store the hashtag with the hex code. But, although it offers convenience and consistency on our side, this would also be a waste of data storage (because the hashtag is implied and would need to be stored `#users`*`#subjects` times). Perhaps it would be better to achieve consistency by omitting the hashtag (what is currently done in firestore), in which case it wouldn't be a breaking change and can be refactored in the future.

Reply:
Yes @benjamin-shen I think refactoring later so that we only have one place where we have subject colors make sense. I agree that omitting the hashtag would make it easier. I'll make a Notion item!